superset/assets/javascripts/explorev2/components/SelectField.jsx

```diff
@@ -31,7 +31,11 @@ export default class SelectField extends React.Component {
     if (this.props.multi) {
       optionValue = opt ? opt.map((o) => o.value) : null;
     }
-    this.props.onChange(this.props.name, optionValue);
+    if (this.props.name === 'datasource' && optionValue !== null) {
+      this.props.onChange(this.props.name, optionValue, opt.label);
```

Comment:
we require datasource_name arg in saveOrOverwriteSlice in views.py, to be used in SaveModal